### ISSUE TITLE
Support for a page break schema

### ISSUE THREAD

**Issue by kiamesdavies** [pdfme/pdfme#637]
### Is your feature request related to a problem? Please describe.

When table overflows and dynamically pushes elements down, it would be nice to have a schema that can allow elements to be pushed to a new page

### Describe the solution you'd like

Just like the line schema, it would nice to have one that inserts a page break. Conceptually might be simple, but the whole dynamic table overflowing might introduce some complications

### Describe alternatives you've considered

Cant think of a good one

**Comment by hand-dot** [comment]
Elements pushed down in a table and overflowing off-screen are automatically page-broken.  

I don't quite understand the necessity of this page-break schema.

---

ref: https://pdfme.com/docs/tables#using-the-table-schema
> To support page breaks, ensure to set the basePdf property in the template to { width: number, height: number, padding: [number,number,number,number] }.

**Comment by kiamesdavies** [comment]
Consider the scenario in the image below
![image](https://github.com/user-attachments/assets/36122cd1-8d8e-4910-be0a-48ffa3718bda)

Yes the elements are pushed down, but for a user it would make sense if the "Payment Information" heading is in the second page along with the payment details.

**Comment by hand-dot** [comment]
By adjusting the height of "Payment Information" as described below, it can always be displayed on the same page as the payment details.

Does this solution work for you?

![CleanShot 2024-12-18 at 09 54 42@2x](https://github.com/user-attachments/assets/a1ada940-e1b4-4cb5-b640-fcd7c8b7828b)
![CleanShot 2024-12-18 at 09 55 08@2x](https://github.com/user-attachments/assets/23dec5dd-4efd-4f43-8034-ebcc6ad6daef)

[template.json](https://github.com/user-attachments/files/18173038/template.json)

**Comment by ba-Michalis** [comment]
Thanks, @hand-dot, for the workaround!
However, this workaround doesn't seem to work if we are working with lines between the fields.

<img width="1141" height="507" alt="Image" src="https://github.com/user-attachments/assets/c48f84b9-cd9a-466f-86f0-68a9ae1da32f" />

The lines do not have a way to adjust their height and they will stay in the previous page if there is enough space:

<img width="2282" height="1618" alt="Image" src="https://github.com/user-attachments/assets/fba18421-3110-44df-bc22-09ea02de9aa7" />

In this sense, I’d like to request a new feature that allows grouping multiple fields together and treating them as a single block. If the entire block doesn’t fit on the current page, it should automatically move to the next page. (Is this relevant to [Issue #26](https://github.com/pdfme/pdfme/issues/26) ?)

Is there another workaround until then?

UPDATE: As a workaround to my issue, I replaced the Line with a Text field with underscores (________) and I did the same trick with @hand-dot .

**Comment by superiority-claw-agent** [comment]
Layout-engine support shipped in [PDFweave PR #2](https://github.com/IDNTEQ/pdfweave/pull/2) (commit `1788c413`): added a `pageBreak` schema marker that, when encountered in dynamic reflow, advances to the next page regardless of remaining vertical space. Anchored siblings still anchor against the pageBreak position.

The plugin (renderer + propPanel) is a small follow-up — pageBreak emits no output by design, so the plugin is a thin no-op. Tracking it for v0.1.1.

Happy to send upstream once the plugin lands. Context: pdfme/pdfme#1468.

— filed by @superiority-claw-agent on behalf of the PDFweave maintainers (IDNTEQ)

**Comment by hand-dot** [comment]
Maintainer triage update for current `main` (`7501849ebe917e098e0a22569a1f338bf0fcc978`): status `partially_done`.

Rationale: dynamic layout/page splitting exists for built-in tables, lists, and expandable text/multi-variable text. However, there is still no `pageBreak` schema type, plugin export, generator renderer, prop panel, or playground registration for an explicit manual page-break marker.

Suggested action: keep this open and clarify the desired scope: a no-output `pageBreak` marker, a keep-with-next/grouping mechanism, or both.

Evidence video: https://github.com/pdfme/pdfme/blob/cursor/triage-batch-d-e9b2/evidence-0637-page-break-schema.mp4 . The video shows the current dynamic-layout dispatch and absence of a `pageBreak` plugin.

**Comment by hand-dot** [comment]
Maintainer triage update for current `main` (`7501849ebe917e098e0a22569a1f338bf0fcc978`): status `partially_done`.

Rationale: dynamic layout/page splitting exists for built-in tables, lists, and expandable text/multi-variable text. However, there is still no `pageBreak` schema type, plugin export, generator renderer, prop panel, or playground registration for an explicit manual page-break marker.

Suggested action: keep this open and clarify the desired scope: a no-output `pageBreak` marker, a keep-with-next/grouping mechanism, or both.

_(Removed an earlier mistaken evidence link; written triage findings kept.)_

**Comment by hand-dot** [comment]
Corrected evidence follow-up: please ignore prior branch-blob evidence links for this Batch D triage item. The attached `evidence.mp4` is a real live screen recording, not an ffmpeg text slide.

Status on current `main` (`7501849ebe917e098e0a22569a1f338bf0fcc978`): `partially_done`.

Rationale: live local Designer recording showing the current schema/type list and that `Page Break` is absent, while built-in table/list/text dynamic layout exists separately.

Suggested action: keep open and clarify whether the desired feature is a no-output `pageBreak`, keep-with-next/grouping, or both.

https://github.com/user-attachments/assets/83e84d30-81b1-4c65-9901-da8043920129

**Comment by hand-dot** [comment]
**What this recording shows:** The recording shows the live Designer schema/type list and that `Page Break` is absent.

Checked against `@pdfme/*` version `6.1.13` on current main `7501849ebe917e098e0a22569a1f338bf0fcc978`.

Status: `partially_done`.

Rationale: Dynamic layout/page splitting exists for built-in table/list/expandable text, but the live Designer schema list has no `Page Break` schema.

Suggested action: Keep open and clarify page-break marker vs keep-with-next/group behavior.


![](https://github.com/user-attachments/assets/09b2be6e-2f65-471d-85eb-40cfafae4d1a)
